src/hotspot/share/runtime/os.cpp

```diff
@@ -1121,6 +1122,31 @@ void os::print_environment_variables(outputStream* st, const char** env_list) {
   }
 }
 
+void os::print_jvmti_agent_info(outputStream* st) {
+#if INCLUDE_JVMTI
+  // should return all kinds of JVMTI agents, but no xrun agents
+  const JvmtiAgentList::Iterator it = JvmtiAgentList::agents();
+  bool first_agent = true;
+  while (it.has_next()) {
+    const JvmtiAgent* agent = it.next();
+    if (agent != nullptr) {
+      if (first_agent) st->print_cr("JVMTI agents:");
```

Comment:
if no agents are loaded, I would print "JVMTI agents: none" unconditionally. Makes it more obvious than just a missing entry; that could be also an error.